server/src/main/java/org/opensearch/search/aggregations/bucket/terms/IncludeExclude.java

```diff
@@ -584,26 +577,54 @@ public boolean isPartitionBased() {
         return incNumPartitions > 0;
     }
 
-    private Automaton toAutomaton() {
-        Automaton a = null;
+    private Automaton toAutomaton(@Nullable IndexSettings indexSettings) {
+        int maxRegexLength = indexSettings == null ? -1 : indexSettings.getMaxRegexLength();
+        Automaton a;
         if (include != null) {
-            a = include.toAutomaton();
+            if (include.length() > maxRegexLength) {
+                throw new IllegalArgumentException(
```

Comment:
The exception text is identical in both cases, right? Maybe an opportunity to have a static helper like `validateRegexLength(String regex, IndexSettings settings)` or similar

Reply:
https://github.com/opensearch-project/OpenSearch/pull/2814